spec/services/streaming_service_provider_data_service_spec.rb

```diff
@@ -0,0 +1,315 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe StreamingServiceProviderDataService do
+  describe 'get_providers' do
+    let(:movie_title) { 'Fake' }
+    let(:movie) { build(:movie, title: movie_title, release_date: '2023-01-01') }
+    let(:movie_args) do
+      {
+        tmdb_id: movie.tmdb_id,
+        title: movie.title,
+        media_type: 'movie',
+        media_format: 'movie',
+        release_date: movie.release_date
+      }
+    end
+
+    describe 'when the API does not return any data' do
+      it 'returns our default providers with a pay_model of "try"' do
+        allow(Tmdb::Client).to receive(:request).and_return(nil)
+
+        results = StreamingServiceProviderDataService.get_providers(movie_args)
+        aggregate_failures "all default providers appear with 'try'" do
+          expect(results.find {|r| r[:name] == 'Netflix'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'Amazon Prime Video'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'Amazon Video'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'YouTube'}[:pay_model] ).to eq('try')
+          expect(results.find {|r| r[:name] == 'Vudu'}[:pay_model] ).to eq('try')
+        end
+      end
+    end
+
+    describe 'when the API does return results' do
+      describe 'but the results do not have free, flatrate, rent, or buy pay model options' do
+        it 'returns our default providers with a pay_model of "try"' do
+          response_data = {
+            results: {
+              US: {
```

Comment:
not sure if this is related to the `watch_region question above.

Reply:
Sort of. I assume if they filtered out the results by our chosen region for us, I wouldn't have to dig for it and it wouldn't look like this in the specs. Alas, their filtering does not work, so I dig for it.